### PR TITLE
Upgrade to go 1.10

### PR DESCRIPTION
**Pull request by derekwbrown**
Try with dummy file

Add pkg_config_path to source tests

Pick specific test docker image for testing with go1.10.2

Change both tests to go110

### CODE REVIEW

File: pkg/util/xc/dummy.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package xc
+
+func doNothing() int64 {
+	return 0
+}
```
Comment (olivielpeau) [inline pkg/util/xc/dummy.go]: probably a temp file you can remove now :)

**Review by olivielpeau — COMMENTED** [review]
Just one nit

The fixes on the logger are also part of this PR: https://github.com/DataDog/datadog-agent/pull/1758

Looks like the gitlab pipeline is failing on the dogstatsd size test because the binary is now 18MB large (instead of ~13MB with go 1.9), probably interesting to look into it quickly and sync with Kenafeh on this.

File: pkg/util/xc/dummy.go (hunk 1 above)
Comment (derekwbrown) [reply pkg/util/xc/dummy.go]: No.  go vet complains that pkg/util/xc doesn't contain any files otherwise.

**Review by olivielpeau — COMMENTED** [review]
A few more comments, once addressed the PR LGTM 👍

**Pull request merged**